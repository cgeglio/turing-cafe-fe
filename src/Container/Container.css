section {
  align-items: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 10px;
  width: 80%;
}

.sorter {
  margin: 10px 40%;
}

.sort-btn {
  border: 3px solid #028080;
  background-color: #028080;
  color: white;
  cursor: pointer;
  margin-top: 5px;
  width: 70%;
}

.sort-btn:hover {
  border: 3px solid #028080;
  background-color: white;
  color: #028080;
}
