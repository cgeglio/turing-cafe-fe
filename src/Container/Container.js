import React from 'react';
import './Container.css';
import Card from '../Card/Card.js'

const Container = ({ reservations, addReservation }) => {
  return reservations.map(reservation => {
    return <Card reservation={reservation} addReservation={addReservation}/>
  })
}

export default Container;
