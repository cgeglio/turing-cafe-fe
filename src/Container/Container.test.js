import React from 'react';
import ReactDOM from 'react-dom';
import Container from './Container';
import { shallow } from 'enzyme';

describe('Container', () => {
  let wrapper;
  let mockReservations;
  let mockRemoveReservation = jest.fn();

  beforeEach(() => {
    mockReservations = [{
        id: 1,
        name: "Christie",
        date: "12/29",
        time: "7:00",
        number: 12
    }]
    wrapper = shallow(<Container
      reservations={mockReservations}
      removeReservation={mockRemoveReservation}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

});
