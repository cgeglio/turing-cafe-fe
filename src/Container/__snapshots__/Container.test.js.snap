// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<section>
  <form
    className="sorter"
  >
    <select
      name="order"
      onChange={[Function]}
    >
      <option />
      <option
        value="earliest"
      >
        Earliest to Latest
      </option>
      <option
        value="latest"
      >
        Latest to Earliest
      </option>
    </select>
    <button
      className="sort-btn"
      onClick={[Function]}
    >
      Sort Reservations
    </button>
  </form>
  <Card
    key="Christie-12/29-"
    removeReservation={[MockFunction]}
    reservation={
      Object {
        "date": "12/29",
        "id": 1,
        "name": "Christie",
        "number": 12,
        "time": "7:00",
      }
    }
  />
</section>
`;
